docs/yul_language/functions.rst

```diff
@@ -0,0 +1,31 @@
+Functions
+=========
+
+Functions in Yul cannot access any variable outside their scope, i.e. they can
+only operate with the variables they receive as arguments. You can define types for arguments and returns. If no
+type is specified, the compiler will default to ``u256``.
+
+.. code-block:: yul
+
+    {
+        function noArgs() -> ret
+        {
+            ret := 2
+        }
+
+        function sum(a, b) -> ret : u64
+        {
+            ret := add(b, a)
+        }
+
+        function getMod(c : s32, d : u128, e) -> ret1, ret2 : u64
+        {
+            ret1 := mulmod(c, d, e)
+            ret2 := addmod(c, d, e)
+        }
+
+        function noReturns(a, b)
+        {
+            store(a, b)
+        }
+    }
```

Comment:
This example does not show how it would be used in solidity or how to do function calls.

Reply:
This section is about Yul, not inline assembly, so I should not mention Solidity here. There interaction of Yul and Solidity is described at `inline_assembly.rst`.